lib/resque/data_store.rb

```diff
@@ -91,10 +92,16 @@ def all_resque_keys
     end
 
     def server_time
-      time, _ = @redis.time
+      time, _ = @redis_time_available ? @redis.time : Time.now
       Time.at(time)
     end
 
+    def redis_time_available?
+      @redis.time
```

Comment:
I think `@redis.respond_to?(:time)` will only work if Redis isn't namespaced. Otherwise we would need to check `@redis.redis.respond_to?(:time)`, and I figured it would be more effective to just try to call `time` and let it fail if it doesn't exist.